pandas/core/algorithms.py

```diff
@@ -1678,9 +1678,23 @@ def map_array(
     if not len(arr):
         return arr.copy()
 
-    # we must convert to python types
-    values = arr.astype(object, copy=False)
+    na_value = None
+    mask = isna(arr)
+    if isinstance(arr.dtype, BaseMaskedDtype) and na_action is None:
+        arr = cast("BaseMaskedArray", arr)
+        values = arr._data
+        if arr._hasna:
+            na_value = arr.dtype.na_value
+    else:
+        # we must convert to python types
+        values = arr.astype(object, copy=False)
+
     if na_action is None:
-        return lib.map_infer(values, mapper)
+        return lib.map_infer(
```

Comment:
What happens when the called mapping returns `pd.NA`? I suspect you get NumPy object dtype back as a result.

Reply:
You're right, but strangely, it depends of the input dtype, for example with `Float64` and `Float64[pyarrow]`:

Case 1) - `Float64` in input / `object` in output
```
s = Series([pd.NA, 11, 22, pd.NA], dtype="Float64")
result = s.map(lambda x: x if x is pd.NA else 2 * x)
0    <NA>
1    22.0
2    44.0
3    <NA>
dtype: object
```
In this case data are converted to a `float64` dtype before the `lib.map_infer` call.

Case 2) - `Float64[pyarrow]` input / `float64` in output
```
s = Series([pd.NA, 11, 22, pd.NA], dtype="Float64[pyarrow]")
result = s.map(lambda x: x if x is pd.NA else 2 * x)
0     NaN
1    22.0
2    44.0
3     NaN
dtype: float64
```
In this case data are converted to an `object` dtype before the `lib.map_infer` call.

Maybe it is linked to `maybe_convert_objects` in `lib.map_infer` that does not identify float value in float64 type ? (with the `seen.float_` flag) or maybe it is linked to a basic error that I did not identify yet :-)

(I will investigate this problem)

Reply:
For information I'm struggling a bit, but I think that I have almost a solution :-)